linode/manager
# refactor: [M3-5181] - RQ-ify Events

## Description 📝
Part 1: [NotificationMenu, EventsLanding (#9046)](https://github.com/linode/manager/pull/9046)
Part 2: [Event Handlers (#9293)](https://github.com/linode/manager/pull/9293)

Migrates the Redux Events store to RQ with the new `useEventsInfiniteQuery` hook.

## What exactly do Events do ❓ *(Copied from Part 2)*
Events have a few different functions in Cloud Manager:

* Notify the user of ongoing/completed changes and updates to their account entities (Linodes, NodeBalancers, etc.)
    * These Event notifications are displayed in the Notification Menu and the Events Landing at /events.
    * Toast notifications appear when long running tasks complete (i.e., a Linode migration)
* Trigger event handlers that update locally cached data
    * E.g., when a Linode is updated in another tab, the Linodes cache is invalidated and Cloud Manager requests and displays the latest details without having to refresh the page
* Display the current status of an entity
    * E.g., when a Linode resize is in progress, this information, along with the progress will be displayed in the Linode details page

## Major Changes 🔄
- New `useEventsInfiniteQuery` hook to handle fetching events and registering event callback handlers
- New `withEventsInfiniteQuery` HOC to enable class-based components to use events
- Refactor `EventsLanding` and `NotificationMenu` components to fetch and display events using the new hook
- Consolidate events handlers in `useAppEventHandlers` and `useToastNotifications` hooks in `App.tsx`
- Remove `rxjs` as a dependency
- Delete the Events Redux store and all associated files (reducers, actions, helpers, etc.)

## How to test 🧪
1. Follow the [testing steps in Part 1](https://github.com/linode/manager/pull/9046#:~:text=How%20to%20test,%F0%9F%A7%AA) to ensure **`EventsLanding` and `NotificationMenu`** work as expected and there are no regressions from prod.
2. Follow the [testing steps in Part 2](https://github.com/linode/manager/pull/9293#:~:text=How%20to%20test,%F0%9F%A7%AA) to ensure **event handlers and toasts** work as expected and there are no regressions from prod.
3. Verify all unit tests and E2E tests pass.
